### DISCUSSION TITLE
I have a monaco-react editor. How can i programatically bring a line in visibility and change the line's background color?

### DISCUSSION THREAD

**Discussion by mushahidkhan835** [suren-atoyan/monaco-react#576] [Q&A]
I have a monaco-react editor. How can i programatically bring a line in visibility and change the line's background color?

**Comment by Rahul5546-X** [comment]
@mushahidkhan835  To bring a range in center you can use methods on editor like revealRangeInCenter

Something like this

`editorRef.current!.revealRangeInCenter(selection, monacoRef.current!.editor.ScrollType.Smooth);`

Take a look at these options, they might help you

https://microsoft.github.io/monaco-editor/typedoc/interfaces/editor.IEditor.html

To change the line's background I think you have to create a customTheme using defineTheme and setTheme or you can apply line decorations(it'll be more easy then creating a custom theme).
